### PR TITLE
Add Button Type (3)

### PR DESCRIPTION
**Pull request by roderickdevelopers**
This should fix all instances of <button> to <button type="button"> or <button type="submit">. 
Can someone look this over and either DM me on Discord or message me here. Thank you

### CODE REVIEW

**Comment by ijlee2** [comment]
@roderickdevelopers I think all examples in `guides/release/in-depth-topics/patterns-for-actions.md` will have `type="button"`. We can guess this because they are not inside a `<form>` element.

If there is a submit and cancel button inside `<form>`, I believe the cancel button has the type of `"button"`.

**Comment by roderickdevelopers** [comment]
@ijlee2 ok I was not sure about that one. I will wait a bit for some more feedback and then fix everything at once.

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 1]
```diff
@@ -13,14 +13,14 @@ We'll call this the `ButtonWithConfirmation` component. We can start off with a
 normal component definition, like we've seen before:
 
 ```handlebars {data-filename=app/components/button-with-confirmation.hbs}
-<button>{{@text}}</button>
+<button type="button">{{@text}}</button>
 
 {{#if this.isConfirming}}
   <div class="confirm-dialog">
-    <button class="confirm-submit">
+    <button type="submit" class="confirm-submit">
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:20]:
```suggestion
    <button type="button" class="confirm-submit">
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 2]
```diff
@@ -13,14 +13,14 @@ We'll call this the `ButtonWithConfirmation` component. We can start off with a
...
       OK
     </button>
-    <button class="confirm-cancel">
+    <button type="submit" class="confirm-cancel">
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:23]:
```suggestion
    <button type="button" class="confirm-cancel">
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 3]
```diff
@@ -63,16 +63,16 @@ export default class ButtonWithConfirmationComponent extends Component {
 ```
 
 ```handlebars
-<button {{on "click" this.launchConfirmationDialog}}>
+<button type="submit" {{on "click" this.launchConfirmationDialog}}>
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:66]:
```suggestion
<button type="button" {{on "click" this.launchConfirmationDialog}}>
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 4]
```diff
@@ -63,16 +63,16 @@ export default class ButtonWithConfirmationComponent extends Component {
...
   {{@text}}
 </button>
 
 {{#if this.isConfirming}}
   <div class="confirm-dialog">
-    <button class="confirm-submit">
+    <button type="submit" class="confirm-submit">
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:72]:
```suggestion
    <button type="button" class="confirm-submit">
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 5]
```diff
@@ -63,16 +63,16 @@ export default class ButtonWithConfirmationComponent extends Component {
...
       OK
     </button>
-    <button class="confirm-cancel">
+    <button type="submit" class="confirm-cancel">
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:75]:
```suggestion
    <button type="button" class="confirm-cancel">
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 6]
```diff
@@ -109,19 +109,19 @@ export default class ButtonWithConfirmationComponent extends Component {
 ```
 
 ```handlebars
-<button {{on "click" this.launchConfirmationDialog}}>
+<button type="submit" {{on "click" this.launchConfirmationDialog}}>
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:112]:
```suggestion
<button type="button" {{on "click" this.launchConfirmationDialog}}>
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 7]
```diff
@@ -109,19 +109,19 @@ export default class ButtonWithConfirmationComponent extends Component {
...
   {{@text}}
 </button>
 
 {{#if this.isConfirming}}
   <div class="confirm-dialog">
-    <button
+    <button type="submit"
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:118]: I am reformatting here to match the rest of the attributes
```suggestion
    <button
      type="button"
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 8]
```diff
@@ -109,19 +109,19 @@ export default class ButtonWithConfirmationComponent extends Component {
...
       class="confirm-submit"
       {{on "click" this.submitConfirm}}
     >
       OK
     </button>
-    <button
+    <button type="submit"
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:124]: Same as previous suggestion!
```suggestion
    <button
      type="submit"
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 9]
```diff
@@ -368,21 +368,21 @@ be used in block form and we will pass `confirmValue` as a
...
+<button type="button" {{on "click" this.launchConfirmDialog}}>
   {{this.text}}
 </button>
 
 {{#if this.isConfirming}}
   <div class="confirm-dialog">
     {{yield this.confirmValue}}
 
-    <button
+    <button type="submit"
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:379]:
```suggestion
    <button type="button"
```

File: guides/release/in-depth-topics/patterns-for-actions.md [hunk 10]
```diff
@@ -368,21 +368,21 @@ be used in block form and we will pass `confirmValue` as a
...
       class="confirm-submit"
       {{on "click" this.submitConfirm}}
     >
       OK
     </button>
-    <button
+    <button type="submit"
```
Comment (locks) [inline guides/release/in-depth-topics/patterns-for-actions.md:385]:
```suggestion
    <button type="button"
```

**Comment by locks** [comment]
The rest seems good!

**Comment by ijlee2** [comment]
There's multiline suggestion, but I haven't tried/figured out how to use it yet. 🤔

**Comment by locks** [comment]
@ijlee2 I believe it only works on desktop, you click the + to add a suggestion and continue holding and drag down the lines you want to cover!

**Comment by roderickdevelopers** [comment]
I just batch committed your suggestions.

**Comment by locks** [comment]
Thanks once again for your continued help!

**Comment by roderickdevelopers** [comment]
Awesome!! Glad I can help!

**Pull request merged**